exo/networking/grpc/grpc_discovery.py

```diff
@@ -107,26 +146,47 @@ async def on_listen_message(self, data, addr):
             peer_id = message['node_id']
             peer_host = addr[0]
             peer_port = message['grpc_port']
+            new_priority = message['priority']
             device_capabilities = DeviceCapabilities(**message['device_capabilities'])
-            if peer_id not in self.known_peers:
-                self.known_peers[peer_id] = (GRPCPeerHandle(peer_id, f"{peer_host}:{peer_port}", device_capabilities), time.time(), time.time())
-                if DEBUG_DISCOVERY >= 2: print(f"Discovered new peer {peer_id} at {peer_host}:{peer_port}")
-            self.known_peers[peer_id] = (self.known_peers[peer_id][0], self.known_peers[peer_id][1], time.time())
+            
+            if peer_id in self.known_peers:
+                _, _, _, current_priority = self.known_peers[peer_id]
+                if new_priority > current_priority:
+                    await self.reconnect_peer(peer_id, peer_host, peer_port, new_priority, device_capabilities)
```

Comment:
discovery is not responsible for connecting to peers, it just discovers them. connecting should happen in the `standard_node.py`

Reply:
this requires a bit of a refactor to do properly, I'm focusing on linux fixes right now but if you have any questions async, leave a comment on this PR and tag me